Makefile

```diff
@@ -9,19 +9,17 @@ all: nvim
 
 CMAKE_PRG ?= $(shell (command -v cmake3 || echo cmake))
 CMAKE_BUILD_TYPE ?= Debug
-CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE)
-# Extra CMake flags which extend the default set
-CMAKE_EXTRA_FLAGS ?=
+CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE) $(CMAKE_FLAGS)
```

Comment:
@KillTheMule thanks for going the distance with this! One thing that held me back up to now is that we need to have some back-compat support for `CMAKE_EXTRA_FLAGS` in the Makefile, because tons of people depend on it. So it will remain a "hidden" feature but we'll remove it from docs. 

So keep the line as you changed it, but also we need the old line:

```
CMAKE_EXTRA_FLAGS ?=
CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE) $(CMAKE_FLAGS)
```